### PR TITLE
Fix Orca assertion for USING with different types

### PR DESCRIPTION
**Pull request by dkovalev1**
Fix Orca assertion for USING with different types.

If columns of join USING condition have different types, parser represents them
as RelabelNode to indicate that these columns probably need type adjustment.
Orca failed to process such join conditions due to assert restrictions during
DXL translation of join expression.

This patch allows RelabelNode to be processed by relaxing the assertion. This is
safe and no additional changes are required because Orca is able to handle nodes
of RelabelType

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/70276

File: src/test/regress/sql/with_clause.sql [hunk 1]
```diff
@@ -997,3 +997,22 @@ drop table t_repl_seg2;
...
+-- start_ignore
+drop table if exists tbl2;
+drop table if exists tbl1;
+-- end_ignore
+
+create table tbl2 (a numeric, b varchar(255), c numeric) distributed by(a);
+create table tbl1 (a bigserial, b varchar(15), c varchar(15)) distributed by(a);
+explain (analyze off, costs off, verbose off)
+with
+t1 as (select * from tbl1),
```
Comment (bimboterminator1) [inline src/test/regress/sql/with_clause.sql]: Ready to approve. An we can simplify the test
```
create table tbl1 (b varchar(15)) distributed by(b);
create table tbl2 (b varchar(255)) distributed by(b);
explain (analyze off, costs off, verbose off)
select * from tbl1 join tbl2  using (b); 
```

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/70438

File: src/test/regress/sql/join_gp.sql [hunk 2]
```diff
@@ -867,3 +867,19 @@ select * from t2 join lateral
 
 drop table t1;
 drop table t2;
+
+
+-- Check that USING with different types and CTE is properly planned by ORCA
```
Comment (bimboterminator1) [inline src/test/regress/sql/join_gp.sql]: CTE?

Comment (dkovalev1) [reply src/test/regress/sql/join_gp.sql]: Sorry, mistyped

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/70464

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1370420

**Review by bimboterminator1 — DISMISSED** [review]

File: src/test/regress/sql/with_clause.sql [hunk 3]
```diff
@@ -997,3 +997,4 @@ drop table t_repl_seg2;
 drop table t_repl;
 drop table with_dml_dr_seg2;
 drop table with_dml_dr;
+
```
Comment (RekGRpth) [inline src/test/regress/sql/with_clause.sql]: remove this line

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/70474

**Comment by RekGRpth** [comment]
> Let ORCA process join condition with expressions of RelabelType

change this to

> Fix ORCA assertion for USING with different types

File: src/test/regress/sql/join_gp.sql [hunk 4]
```diff
@@ -867,3 +867,19 @@ select * from t2 join lateral
...
+-- Check that USING with different types is properly planned by ORCA
+-- start_ignore
+drop table if exists tbl2;
+drop table if exists tbl1;
+-- end_ignore
+
+create table tbl1 (b varchar(15)) distributed by(b);
+create table tbl2 (b varchar(255)) distributed by(b);
+
+explain (analyze off, costs off, verbose off)
```
Comment (RekGRpth) [inline src/test/regress/sql/join_gp.sql]:
```suggestion
explain (costs off)
```

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/70481

File: src/test/regress/sql/join_gp.sql (hunk 4 above)
Comment (dkovalev1) [reply src/test/regress/sql/join_gp.sql]: Done

File: src/test/regress/sql/join_gp.sql [hunk 5]
```diff
@@ -867,3 +867,19 @@ select * from t2 join lateral
...
+-- start_ignore
+drop table if exists tbl2;
+drop table if exists tbl1;
+-- end_ignore
+
+create table tbl1 (b varchar(15)) distributed by(b);
+create table tbl2 (b varchar(255)) distributed by(b);
+
+explain (costs off)
+select * from tbl1 join tbl2  using (b);
```
Comment (RekGRpth) [inline src/test/regress/sql/join_gp.sql]:
```suggestion
select * from tbl1 join tbl2 using (b);
```

**Comment by BenderArenadata** [comment]
Failed job Build for ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1381261

**Comment by BenderArenadata** [comment]
Failed job Build for x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1381259

**Comment by BenderArenadata** [comment]
Failed job Build for ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1381445

**Comment by BenderArenadata** [comment]
Failed job Build for x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1381449

**Comment by BenderArenadata** [comment]
Failed job Build for ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1391014

**Comment by BenderArenadata** [comment]
Failed job Build for x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1391013

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/70752

**Comment by BenderArenadata** [comment]
Failed job Behave tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1391427

**Review by RekGRpth — APPROVED** [review]

**Review by bimboterminator1 — APPROVED** [review]

**Pull request merged**